src/Acuminator/Acuminator.Analyzers/Analyzers/DAC/DacMethodsUsageAnalyzer.cs

```diff
@@ -0,0 +1,86 @@
+            ImmutableArray.Create
+            (
+                Descriptors.PX1031_DacCannotContainInstanceMethods,
+                Descriptors.PX1032_DacPropertyCannotContainMethodInvocations
+            );
+
+        internal override void AnalyzeCompilation(CompilationStartAnalysisContext compilationStartContext, PXContext pxContext)
+        {
+            compilationStartContext.RegisterSyntaxNodeAction(syntaxContext => AnalyzeDacDeclaration(syntaxContext, pxContext), SyntaxKind.ClassDeclaration);
+        }
+
+        private void AnalyzeDacDeclaration(SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            ClassDeclarationSyntax classDeclaration = syntaxContext.Node as ClassDeclarationSyntax;
+            INamedTypeSymbol typeSymbol = syntaxContext.SemanticModel.GetDeclaredSymbol(classDeclaration, syntaxContext.CancellationToken);
+
+            if (typeSymbol != null && IsDac(typeSymbol, pxContext))
+            {
+                IEnumerable<MethodDeclarationSyntax> dacMethods = classDeclaration.DescendantNodes().OfType<MethodDeclarationSyntax>();
+
+                foreach (SyntaxNode node in classDeclaration.DescendantNodes())
+                {
+                    if (syntaxContext.CancellationToken.IsCancellationRequested)
+                    {
+                        return;
+                    }
+
+                    if (node is MethodDeclarationSyntax method)
+                    {
+                        AnalyzeMethodDeclarationInDac(method, syntaxContext, pxContext);
+                    }
+                    else if (node is PropertyDeclarationSyntax property)
+                    {
+                        AnalyzeMethodInvocationInDacProperty(property, syntaxContext, pxContext);
+                    }
+                }
+            }
+        }
+
+        private void AnalyzeMethodInvocationInDacProperty(PropertyDeclarationSyntax property, SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            IEnumerable<InvocationExpressionSyntax> methodInvocations = property.DescendantNodes()
+                                                                        .OfType<InvocationExpressionSyntax>();
+            foreach (InvocationExpressionSyntax invocation in methodInvocations)
+            {
+                SymbolInfo symbol = syntaxContext.SemanticModel.GetSymbolInfo(invocation, syntaxContext.CancellationToken);
+
+                if (symbol.Symbol.Kind == SymbolKind.Method)
+                {
+                    syntaxContext.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1032_DacPropertyCannotContainMethodInvocations, invocation.GetLocation()));
+                }
+            }
+        }
+
+        private void AnalyzeMethodDeclarationInDac(MethodDeclarationSyntax method, SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            bool isStatic = method.Modifiers.Any(m => m.IsKind(SyntaxKind.StaticKeyword));
```

Comment:
@metallmonster, could you please move this method as extension method to Acuminator.Utils MethodDeclarationUtils class. It could be usefull in other places